### PR TITLE
Update to GCC 11

### PR DESCRIPTION
**Pull request by bdice**
This PR updates builds to use GCC 11.

### CODE REVIEW

**Comment by bdice** [comment]
Depends on #3309 and #3306. After those merge, CI should pass.

This is ready for review, but please **_do not merge_** until all RAPIDS repos are ready to migrate. (I do not have privileges to add the DO NOT MERGE label to prohibit merging.)

**Review by jjacobelli — APPROVED** [review]

**Pull request merged**
